include/zephyr/drivers/sent/sent.h

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Single Edge Nibble Transmission (SENT) driver API.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_SENT_H_
+#define ZEPHYR_INCLUDE_DRIVERS_SENT_H_
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief SENT Interface
+ * @defgroup sent_interface SENT Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+/**
+ * @brief SENT frame structure
+ */
+struct sent_frame {
+	union {
+		/* ID-Data field for SENT slow frame */
+		struct {
+			uint16_t id;
+			uint16_t data;
+		} slow_frame;
+
+		/* Data field for SENT fast frame */
+		uint32_t data;
+	};
+	uint32_t timestamp;
+	uint8_t crc;
+};
+
+/**
+ * @brief SENT status
+ */
+enum sent_status {
+	SENT_RX_SLOW_FRAME,
+	SENT_RX_FAST_FRAME,
+	SENT_RX_ERR,
+};
+
+/**
+ * @brief Defines the application callback handler function signature for receiving.
+ *
+ * @param dev        Pointer to the device structure for the driver instance.
+ * @param channel    The hardware channel of the driver instance.
+ * @param frame      Received frame.
+ * @param status     SENT frame status.
+ * @param user_data  User data provided when the filter was added.
+ */
+typedef void (*sent_rx_callback_t)(const struct device *dev, uint8_t channel,
+				   struct sent_frame *frame, enum sent_status status,
+				   void *user_data);
+
+/** @cond INTERNAL_HIDDEN */
+
+/**
+ * @brief Callback API upon starting receive SENT
+ * See @a sent_start_rx() for argument description
+ */
+typedef int (*sent_start_rx_t)(const struct device *dev, uint8_t channel);
+
+/**
+ * @brief Callback API upon stopping receive SENT
+ * See @a sent_stop_rx() for argument description
+ */
+typedef int (*sent_stop_rx_t)(const struct device *dev, uint8_t channel);
+
+/**
+ * @brief Callback API upon adding RX callback
+ * See @a sent_add_rx_callback() for argument description
+ */
+typedef int (*sent_add_rx_callback_t)(const struct device *dev, uint8_t channel,
+				      sent_rx_callback_t callback, void *user_data);
+
+__subsystem struct sent_driver_api {
+	sent_start_rx_t start_rx;
+	sent_stop_rx_t stop_rx;
+	sent_add_rx_callback_t add_rx_callback;
```

Comment:
Having "sent" and "rx" together in the function names can be very confusing and might be misinterpreted. Possible name choices:

- sent_listen() / sent_start_listening() / sent_start_capture()
- sent_stop() / sent_stop_listening() / sent_stop_capture()
- sent_register_callback()

I'm assuming you decided to use `_rx_` because SENT is unidirectional. However, when (and if) SENT-B would be available, this API will probably might need to be extended to support bi-directional communication so the "rx" may not be relevant then.

Reply:
I replaced all names of the `sent_ ` prefix in the variations with `saej2716_`
with APIs: `saej2716_start_rx()`, `saej2716_stop_rx() `and `saej2716_add_rx_callback()`
I will consider more your suggestion